File: MinjiJu/2022-06-24-reverse-bits.cpp
class Solution {
public:
    uint32_t reverseBits(uint32_t n) {
        
        uint32_t res = 0;
        
        // iterate over each bit
        for(int i=0; i<32; i++){
            
            // i-th bit from LSB
            uint32_t bit_curr = (n>>i) & 1;
            
            // left shit and add to LSB
            res = (res<<1) + bit_curr;
        }
        return res;
    }
};

File: Chris/2022-06-24-ReverseBits.py
class Solution:
    def reverseBits(self, n: int) -> int:
        reversedBits = 0;
        for i in range(32):
            if n & 1 == 1:
                reversedBits += (2 ** (31-i))
            n = n >> 1
            
        return reversedBits
        